tests/backup/backup_helper.go

```diff
@@ -5559,3 +5561,172 @@ func UpgradeKubevirt(versionToUpgrade string, workloadUpgrade bool) error {
 	}
 	return nil
 }
+
+// UpdateKDMPConfigMap update the KDMP configMap with give data key and value.
+func UpdateKDMPConfigMap(dataKey string, dataValue string) error {
+	KDMPconfigMap, err := core.Instance().GetConfigMap("kdmp-config", "kube-system")
+	if err != nil {
+		return err
+	}
+	KDMPconfigMap.Data[dataKey] = dataValue
+	_, err = core.Instance().UpdateConfigMap(KDMPconfigMap)
+	if err != nil {
+		return err
+	}
+	log.Infof("updated the KDMP configMap data with key [%s]: value [%s]", dataKey, dataValue)
+	return nil
+}
+
+// CreateNestedDirectoriesWithFilesInPod creates a nested directory structure with files within a specified Pod.
+func CreateNestedDirectoriesWithFilesInPod(podName, podNamespace, basePath string, depth, levels, filesPerDirectory int) error {
```

Comment:
Can we convert this function 2 accept only 2 inputs like this `v1.Pod` and `DirectoryConfig` where `DirectoryConfig` is as follows:

```
type PodDirectoryConfig struct {
	BasePath           string
	Depth              int
	Levels             int
	FilesPerDirectory  int
}
```

In future if you need to add more parameters to the directory structure, you just have to modify the struct instead of changing the function signature.